Sources/swift-format/Frontend/Frontend.swift

```diff
@@ -40,7 +40,7 @@ class Frontend {
       // That way they will be printed out, but we'll continue execution on the valid rules.
       let invalidRules = configuration.rules.filter { !RuleRegistry.rules.keys.contains($0.key) }
       for rule in invalidRules {
-        diagnosticsEngine.emitWarning("Configuration contains an unrecognized rule: \(rule.key)", location: nil)
+        diagnosticsEngine.emitError("Configuration contains an unrecognized rule: \(rule.key)", location: nil)
```

Comment:
I think this could break clients in the following scenario: You have a Swift 6.1 rule added to your `.swift-format` file but locally you are running a 6.0 `swift-format` to format your code (not to lint). That should mostly work but with this change, formatting would exit with 1 and thus fail (IIUC Xcode would show an error message in that case). So, I think this would need to become a print to stderr instead of going through the diagnostic engine.

CC @allevato Do you have thoughts on this?

Reply:
I'll do a more thorough soon, but for this issue specifically: I'd like to keep all messages from the formatter going through the diagnostics engine instead of having some there and some through just `stderr`.

I think the answer here is that we should only remove the severity from `Finding`s but not `Diagnostic`/`DiagnosticEngine`. That removes the complexity from the rules themselves (and forces all findings to become errors, which is the change we're asking for), but still lets the formatter frontend issue other diagnostics that aren't as severe.